controllers/testclustertopology_controller.go

```diff
@@ -124,9 +124,26 @@ func (r *TestClusterTopologyReconciler) Reconcile(req ctrl.Request) (ctrl.Result
 			}
 			return ctrl.Result{}, nil
 		}
-
 		// cluster config no change
 		if !tiup.IsClusterConfigModified(ct.Status.PreTiDBCluster, ct.Spec.TiDBCluster) {
+			var rr naglfarv1.TestResourceRequest
+			if err := r.Get(ctx, types.NamespacedName{
+				Namespace: req.Namespace,
+				Name:      ct.Spec.ResourceRequest,
+			}, &rr); err != nil {
+				return ctrl.Result{}, err
+			}
+			isRequeue, result, err := r.pruneTiDBCluster(ctx, &ct, &rr)
```

Comment:
Source viewers(like me) may don't know why needs do `pruneTiDBCluster` here. Perhaps you should create a new function, or add an if-condition (predicate according to `status` that need to wait for prune finished) around this code or give it a not too bad name to it.